File: vibebot/workers/collector.py
import logging

log = logging.getLogger(__name__)


def run(plugins: list, conn) -> None:
    """Stage 1: Call collect() on each enabled plugin."""
    for plugin in plugins:
        count = plugin.collect(conn)
        log.info("Plugin '%s' collected %d new item(s).", plugin.name, count)

File: src/workers/collector.ts
import Database from 'better-sqlite3';
import { BasePlugin } from '../core/basePlugin';

export async function run(plugins: BasePlugin[], db: Database.Database): Promise<void> {
  for (const plugin of plugins) {
    const count = await plugin.collect(db);
    console.info(`Plugin '${plugin.name}' collected ${count} new item(s).`);
  }
}
